src/service/mercury/helpers/transformers.ts

```diff
@@ -69,6 +100,135 @@ interface TokenDetails {
+  const sellingLiabilities = accountObject?.sellingLiabilities || "0";
+  const nativeBalance = accountObject?.nativeBalance || "0";
+
+  const accountBalance = {
+    native: {
+      token: { type: "native", code: "XLM" },
+      total: formatTokenAmount(new BigNumber(nativeBalance), 7),
+      available: new BigNumber(BASE_RESERVE_MIN_COUNT)
+        .plus(new BigNumber(numSubEntries))
+        .plus(new BigNumber(numSponsoring))
+        .minus(new BigNumber(numSponsored))
+        .times(new BigNumber(BASE_RESERVE))
+        .plus(new BigNumber(sellingLiabilities)),
+    },
+  };
+
+  const classicBalances = accountCurrentTrustlines.reduce((prev, curr) => {
+    const tl = curr;
+    const trustline = xdr.Asset.fromXDR(tl.asset, "base64");
+    switch (trustline.switch().name) {
+      case "assetTypeNative": {
+        // not in this query, in account object query
+        return prev;
+      }
+
+      case "assetTypeCreditAlphanum4": {
+        const code = trustline.alphaNum4().assetCode().toString();
+        const issuer = StrKey.encodeEd25519PublicKey(
+          trustline.alphaNum4().issuer().ed25519()
+        );
+        prev[`${code}:${issuer}`] = {
+          token: {
+            code,
+            issuer: {
+              key: issuer,
+            },
+          },
+          total: formatTokenAmount(new BigNumber(tl.balance), 7),
+          available: formatTokenAmount(new BigNumber(tl.balance), 7),
+        };
+        return prev;
+      }
+
+      case "assetTypeCreditAlphanum12": {
+        const code = trustline.alphaNum12().assetCode().toString();
+        const issuer = StrKey.encodeEd25519PublicKey(
+          trustline.alphaNum12().issuer().ed25519()
+        );
+        prev[`${code}:${issuer}`] = {
+          token: {
+            code,
+            issuer: {
+              key: issuer,
+            },
+          },
+          total: formatTokenAmount(new BigNumber(tl.balance), 7),
+          available: formatTokenAmount(new BigNumber(tl.balance), 7),
+        };
+        return prev;
+      }
+
+      case "assetTypePoolShare": {
+        // Should pool shares be decoded here?
+        return prev;
+      }
+
+      default:
+        throw new Error("Asset type not suppported");
+    }
+  }, {} as NonNullable<AccountBalancesInterface["balances"]>);
+
+  const tokenBalanceData = contractIds.map((id) => {
+    const resData = rawResponseCurrentData?.data || ({} as any);
+    const tokenRecord = resData[id] || [];
+    return tokenRecord;
+  });
+
+  const formattedBalances = tokenBalanceData.map(([entry]) => {
+    const details = tokenDetails[entry.contractId];
+    const valEntry = xdr.LedgerEntry.fromXDR(entry.valXdr, "base64");
+    const val = valEntry.data().contractData().val();
+    return {
+      ...entry,
+      ...details,
+      total: scValToNative(val),
+    };
+  });
+
+  const balances = formattedBalances.reduce((prev, curr) => {
+    prev[`${curr.symbol}:${curr.contractId}`] = {
```

Comment:
Hm, do we need to do something similar to [this](https://github.com/stellar/freighter-backend/pull/59/files#diff-d156b034490e7e62a6108698b05f888aee8e849e455b04c7bb976baf0686b076R585) to make sure we don't get duplicate balances for SAC's?

Reply:
Oh yeah good call, I think we would but I need to set up a SAC balance to be sure so I'll go do that and report back. Thanks!

Reply:
I added a helper for this in 65368b98ca652f0f5c8df3799f926dab58d8900d
After reviewing the convo where this info came from, I do think we need to take this a step further and either compute the SAC address from the name and check for matching contracts or compare the wasm for the contract against the known SAC wasm. This would ensure that tokens that set their name to match our sac issuer pattern would not be filtered out of views.